packages/blocks/src/store/actions.js

```diff
@@ -19,6 +19,36 @@ export function addBlockTypes( blockTypes ) {
 	};
 }
 
+/**
+ * Yields action objects signaling that the passed block type's settings should be stored in the state.
+ *
+ * @param {WPBlockType} blockType Unprocessed block type settings.
+ *
+ * @yield {Object} Action object.
+ */
+export function* __experimentalAddBlockType( blockType ) {
+	yield {
+		type: 'ADD_UNPROCESSED_BLOCK_TYPE',
+		blockType,
+	};
+	yield {
+		type: 'APPLY_BLOCK_TYPE_FILTERS',
```

Comment:
Why are these written as controls instead of being a generator here in actions?

Reply:
I might misunderstood the role of action and controls. I looked at several existing examples where actions yield simple objects and everything else is handled with controls. My understanding is that it help to decouple side effects.

Reply:
yes, that's their role, but for me `APPLY_BLOCK_TYPE_FILTERS` is not really what I call a side effect, it's just "logic" to update the blocks in the state right? For me controls are more about repeatable small things (api request, select, dispatch...)

Reply:
I can move all the logic to the actions file and inline as a generator instead of keeping it in the controls file 👍🏻